lib/events.js

```diff
@@ -435,13 +453,27 @@ function listenerCount(type) {
 }
 
 EventEmitter.prototype.eventNames = function eventNames() {
-  return this._eventsCount > 0 ? Reflect.ownKeys(this._events) : [];
+  return this._events !== undefined && this._events.size ?
+    [...this._events.keys()] : [];
```

Comment:
Did you benchmark this change? Is the spread slower than manually looping and filling an array?

Reply:
It might be, honestly I wasn't sure if this was performance intensive code and it works quite fast as is. Happy to dive in deeper once I get the Proxy issue sorted.